src/query/frontend/ast/ast.hpp

```diff
@@ -1770,7 +1781,7 @@ class NodeAtom : public memgraph::query::PatternAtom {
     return visitor.PostVisit(*this);
   }
 
-  std::vector<memgraph::query::LabelIx> labels_;
+  std::vector<std::variant<memgraph::query::LabelIx, memgraph::query::Expression *>> labels_;
```

Comment:
I would prefer the variant was made into a type with api that handles insertion/deletion. You copy/paste the same bit of code in a few places.